### PR TITLE
Static code analysis using PMD

### PR DESCRIPTION
**Pull request by zod**
Opening files with IntelliJ IDEA shows a lot of warnings for our code. This is distracting and makes it hard to see new warnings being added.

[PMD](https://pmd.github.io/) and the [gradle plugin](https://docs.gradle.org/current/userguide/pmd_plugin.html) allow automatic code checks during build (and CI) to prevent new violations from being added.

I've enabled the [quickstart](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java.html#additional-rulesets) ruleset and initially disabled all rules which are currently violated. Re-enabled (and fixed) rules are:

* [AvoidInstanceofChecksInCatchClause](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_errorprone.html#avoidinstanceofchecksincatchclause)
* [LogicInversion](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_design.html#logicinversion)
* [LooseCoupling](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_bestpractices.html#loosecoupling)
* [SimplifiableTestAssertion](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_bestpractices.html#simplifiabletestassertion)
* [SimplifiedTernary](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_design.html#simplifiedternary)
* [SimplifyBooleanReturns](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_design.html#simplifybooleanreturns)
* [SingularField](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_design.html#singularfield)
* [UnnecessaryFullyQualifiedName](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname)
* [UnnecessaryImport](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_codestyle.html#unnecessaryimport)
* [UnnecessaryModifier](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_codestyle.html#unnecessarymodifier)
* [UnnecessaryReturn](https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_codestyle.html#unnecessaryreturn)

Each rule is a single commit and can be removed I it's deemed to strict.

### CODE REVIEW

**Comment by afischerdev** [comment]
Fine helper.
But before we merge this please merge all pending PRs that have no conflicts.
- #478 and #481
- #456 could be closed, I already have a test based on the same logic as in #474. I add it later.
- #458 can wait, needs a decision
- #188 I could add when I come in lib update on that passage
- #457 is draft

**Comment by zod** [comment]
I'm not sure why the other PRs are a prerequisite to this PR but I've either merged, commented or closes those PRs.

If this PR is merged those PRs that are still open should merge the master to get the ability to run all checks locally instead of getting feedback just from the CI build.

**Review by afischerdev — APPROVED** [review]

**Pull request merged**
